crates/subspace-networking/src/connected_peers.rs

```diff
@@ -219,41 +219,59 @@ impl<Instance> Behaviour<Instance> {
 
     /// Specifies the whether we should keep connections to the peer alive. The decision could
     /// depend on another protocol (e.g.: PeerInfo protocol event handling).
-    /// In case when we had decision timeout it sets up proper keep connection alive anyway.
     pub fn update_keep_alive_status(&mut self, peer_id: PeerId, keep_alive: bool) {
-        let (connection_state, keep_alive) = if keep_alive {
-            if self.permanently_connected_peers() < self.config.target_connected_peers {
-                trace!(%peer_id, %keep_alive, "Insufficient number of connected peers.");
+        let enough_connected_peers =
+            self.permanently_connected_peers() < self.config.target_connected_peers;
+        // It's a known peer.
+        if let Some(connection_state) = self.known_peers.get_mut(&peer_id) {
+            // We're connected
+            if let Some(connection_id) = connection_state.connection_id() {
+                // Check whether we have enough connected peers already
+                let (new_connection_state, keep_alive) = if enough_connected_peers {
```

Comment:
Why not short-circuiting right away at the top returning `(ConnectionState::NotInterested, KeepAlive::No)`? It'd remove one level of nesting and there is really no need to do extra work here.

Reply:
I didn't understand you fully, however,  I simplified the code a bit. Please, have another look.